instrumentation/opentelemetry-instrumentation-logging/src/opentelemetry/instrumentation/logging/__init__.py

```diff
@@ -0,0 +1,201 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+The OpenTelemetry ``logging`` integration automatically injects tracing context into log statements.
+
+The integration patches the standard library logging module to automatically inject tracing context into log record objects
+and optionally calls ``logging.basicConfig()`` to set a logging format with placeholders for span ID, trace ID and service name.
+
+The following keys are injected into the patched log record objects:
+
+- ``otelSpanID``
+- ``otelTraceID``
+- ``otelServiceName``
+
+The integration uses the following logging format by default:
+
+.. code-block::
+
+    %(asctime)s %(levelname)s [%(name)s] [%(filename)s:%(lineno)d] [trace_id=%(otelTraceID)s span_id=%(otelSpanID)s service_name=%(otelServiceName)s] - %(message)s
+
+Enable trace context injection
+------------------------------
+
+The integration is opt-in and must be enabled explicitly by setting the environment variable ``OTEL_PYTHON_LOG_CORRELATION`` to ``true``.
+
+The integration always patches the logging module to inject tracing context into the log record objects. Setting ``OTEL_PYTHON_LOG_CORRELATION`` to ``true``
+calls ``logging.basicConfig()`` to set a logging format that actually makes use of the injected variables.
+
+
+Environment variables
+---------------------
+
+OTEL_PYTHON_LOG_CORRELATION
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This env var must be set to ``true`` in order to enable trace context injection into logs by calling ``logging.basicConfig()`` and
+setting a logging format that makes use of the injected tracing variables.
+
+The default value is ``false``.
+
+OTEL_PYTHON_LOG_FORMAT
+^^^^^^^^^^^^^^^^^^^^^^
+
+This env var can be used to instruct the instrumentation to use a custom logging format.
+
+The default value is:
+
+.. code-block::
+
+    %(asctime)s %(levelname)s [%(name)s] [%(filename)s:%(lineno)d] [trace_id=%(otelTraceID)s span_id=%(otelSpanID)s service_name=%(otelServiceName)s] - %(message)s
+
+
+OTEL_PYTHON_LOG_LEVEL
+^^^^^^^^^^^^^^^^^^^^^
+
+This env var can be used to set a custom logging level.
+
+The default value is ``info``.
+
+Options are:
+
+- ``info``
+- ``error``
+- ``debug``
+- ``warning``
+
+Manually calling logging.basicConfig
```

Comment:
I think the need for implicit ordering like this is a good argument why opentelemetry shouldn't provide a basicConfig configuration via environment variables.

If the user has to remember to call opentelemetry instrumentation first to get the opentelemetry call, why not just expose useful constants (e.g. log format) and document how to use them?

Reply:
We have a constant defined and can document it. We can also update documentation to show how users can do this completely with custom code (manually calling basicConfig) without dealing with environment variables. 

The environment variables solve an important use case though which is 100% auto-instrumented apps where users cannot or don't want to add manual code to enable tracing. We support this with the `opentelemetry-bootstrap` command and the pattern so far to allow customization of behavior has been to use env vars. So we really do need to support this through env vars. 

This can still be totally used manually by adding a bit of code. I'll update the documentation to cover that use case as well.

Reply:
> The environment variables solve an important use case though which is 100% auto-instrumented apps where users cannot or don't want to add manual code to enable tracing. We support this with the opentelemetry-bootstrap command and the pattern so far to allow customization of behavior has been to use env vars. So we really do need to support this through env vars.

A part of me feels like no one would want opentelemetry to expose such an opinionated standard log, but I see your point that the idea is to make it really easy for even beginners to python / application development to get observability. 

SGTM to leave the env vars.